src/core/utils/channel_manager.cpp

```diff
@@ -435,7 +435,7 @@ otError ChannelManager::SetAutoChannelSelectionInterval(uint32_t aInterval)
 
     if (mAutoSelectEnabled && (mState == kStateIdle) && mTimer.IsRunning() && (prevInterval != aInterval))
     {
-        mTimer.StartAt(mTimer.GetFireTime() - TimerMilli::SecToMsec(prevInterval), TimerMilli::SecToMsec(aInterval));
+        mTimer.StartAt(mTimer.GetFireTime(), Timer::NormalizeDt(aInterval - prevInterval));
```

Comment:
Would this work if `aInterval` is smaller than `prevInterval`?

The `aInterval - prevInterval` would be negivate and therefor a large `uint32_t` value (last bit would be set) and `normalize` would set it to `Timer::kMaxDt` which I think won't be correct...?

Suggest keeping as before: We start the timer from last timer start time (which is `fireTime - prevInterval`) for new interval...

Reply:
yes, thank you!